web/portal/brand/src/entities/AdministratorRelPublicEntity/Action/RevokeAll.tsx

```diff
@@ -0,0 +1,142 @@
+import { MoreMenuItem } from '@irontec/ivoz-ui/components/List/Content/Shared/MoreChildEntityLinks';
+import { StyledTableRowCustomCta } from '@irontec/ivoz-ui/components/List/Content/Table/ContentTable.styles';
+import {
+  OutlinedButton,
+  SolidButton,
+} from '@irontec/ivoz-ui/components/shared/Button/Button.styles';
+import {
+  ActionFunctionComponent,
+  isSingleRowAction,
+  MultiSelectActionItemProps,
+} from '@irontec/ivoz-ui/router/routeMapParser';
+import _ from '@irontec/ivoz-ui/services/translations/translate';
+import ReceiptLongIcon from '@mui/icons-material/ReceiptLong';
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Tooltip,
+} from '@mui/material';
+import { useState } from 'react';
+import { useStoreActions, useStoreState } from 'store';
+
+const RevokeAll: ActionFunctionComponent = (
+  props: MultiSelectActionItemProps
+) => {
+  const { rows, selectedValues = [], variant = 'icon' } = props;
+
+  const [open, setOpen] = useState(false);
+  const [error, setError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState<string>();
+
+  const administrators = useStoreState(
+    (state) => state.entities.entities.Administrator
+  );
+
+  const filteredRows = rows.filter((row) => {
+    return selectedValues.includes(row.id.toString());
+  });
+
+  const entityIds = filteredRows.map((item) => item?.publicEntityId?.id);
+  let disabled = true;
+
+  if (selectedValues.length > 0) {
+    disabled = false;
+  }
+
+  const apiPost = useStoreActions((actions) => {
+    return actions.api.post;
+  });
+
+  const handleClickOpen = () => {
+    if (disabled) {
+      return;
+    }
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    setError(false);
+  };
+
+  const administratorId =
+    typeof rows[0]?.administrator === 'object'
+      ? rows[0].administrator.id
+      : null;
+
+  const handleUpdate = () => {
+    const payload = entityIds;
+    const path = `${administrators.path}/${administratorId}/revoke_all`;
+    const resp = apiPost({
+      path,
+      values: payload,
+      contentType: 'application/json',
+      silenceErrors: true,
+    })
+      .then(() => {
+        setOpen(false);
+
+        if (resp !== undefined) {
+          setTimeout(() => {
+            window.location.reload();
+          });
+        }
+      })
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      .catch((error: Record<string, any>) => {
```

Comment:
```suggestion
      .catch((error: ApiError) => {
```
Defined in ivoz-ui/services/api/ApiClient.ts